cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -359,6 +360,8 @@ private void suspendResumeSchedule(HttpResponder responder, String namespaceId,
       }
     } catch (SecurityException e) {
       responder.sendStatus(HttpResponseStatus.UNAUTHORIZED);
+    } catch (NotFoundException e) {
+      responder.sendString(HttpResponseStatus.NOT_FOUND, e.getMessage());
```

Comment:
this is almost good. (below). For unexpected Throwables: Log as error, respond with internal error. The only thing missing is: what schedule was it and did we try to suspend it or resume it?